Persistent Volume Dashboard

njuettner opened the pull request:
My PR improves the old PV dashboard. I also changed the ownership to turtles (formerly ludacris 😅)

<img width="2551" alt="image" src="https://github.com/giantswarm/dashboards/assets/1936982/15037f91-fd2e-430d-aa41-7ae7e5d18792">


Some improvements:

- Detect pending / lost PV's
- Estimation of PV's when hitting 100% based on daily / weekly usage
- Current Disk RW per Volume
- Overview of Storage Classes 

Related issues:

- https://github.com/giantswarm/giantswarm/issues/27226
- https://github.com/giantswarm/roadmap/issues/2536

### Checklist

- [x] Update changelog in CHANGELOG.md in an end-user friendly language.

LolloneS commented:
Nice! ❤️ feel free to mark the task [here](https://github.com/giantswarm/giantswarm/issues/27226) as done once merged 😄

hervenicol reviewed (changes requested):
These updates look great, but the dashboard comes from upstream mixins (https://github.com/kubernetes-monitoring/kubernetes-mixin/tree/master/dashboards), so no changes should be done directly here.

I suggest either you fork and create "our" persistent volumes dashboard, and/or contribute to upstream mixins.

Feel free to ping me on Slack if you want extra info ;)

njuettner commented:
@hervenicol good point, I forgot about the upstream part 👍🏻. 

I moved it into private now and leave the one in mixin untouched.

One thing is I removed management-cluster-pvc.json which is also named `Kubernetes Persistent Volumes` but it does not provide any more information and it only focuses on MC PV's.

<img width="2543" alt="image" src="https://github.com/giantswarm/dashboards/assets/1936982/67a2865e-6247-415d-9fde-3741fe8d3712">

The dasboard I propose does contain the MC PV's as well. So I think we can safely remove it.

hervenicol commented:
Oh yes, I was suspecting there was a kind of conflict, good to remove the one in your screenshot! :+1: 

Make sure you set a different UID and title to yours than the mixins dashboard then. I like to set a custom UID (can be any string) so it's easier to reference from alerts.

njuettner commented:
@hervenicol is that ok?

Mixin:

```
   "title": "Kubernetes / Persistent Volumes",
   "uid": "919b92a8e8041bd567af9edab12c840c",
```

New one:

```
   "title": "Kubernetes Persistent Volumes",
   "uid": "113b92a8e8041bd567af9edab12c374k",
```

hervenicol commented on helm/dashboards/dashboards/shared/private/persistentvolume.json:
```diff
@@ -0,0 +1,1586 @@
+   },
+   "timepicker": {},
+   "timezone": "utc",
+   "title": "Kubernetes Persistent Volumes",
+   "uid": "119b92a8e8041bd567af9edab12c840c",
```
Currently, this enough to avoid conflicts with the mixins dashobard.
But I think we can do even better.

You can set human-readable UID, which will ease when you link the dashboard in alerts.
And I would emphasis in the title that this dashboard is a GS one, as opposed to the mixins one.
```suggestion
   "title": "Giantswarm / Kubernetes Persistent Volumes",
   "uid": "GSPersistentVolumes",
```

WDYT?

njuettner replied:
this makes sense to me 👍🏻

hervenicol reviewed (approved):
Looks great :heart:
